app/src/main/java/com/group80/uoftinder/UpdateList.java

```diff
@@ -0,0 +1,51 @@
+package com.group80.uoftinder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class UpdateList extends User {
+    private User currentUser; // the current user
+    private User displayedUser; // the user that is displayed to current user
+    private boolean liked; // true if currentUser 'likes' displayedUser, false otherwise
+    private List<User> viewedList; // currentUser's viewedList
+    private List<User> likedList; // currentUser's likedList
+
+    /**
+     * Constructor for UpdateList Class.
+     * @param displayedUser is the user that is currently being displayed to currentUser.
+     * @param liked is True when currentUser 'likes' displayedUser, False otherwise.
+     */
+    public UpdateList(User currentUser, User displayedUser, boolean liked) {
+        super();
+        this.currentUser = currentUser;
+        this.displayedUser = displayedUser;
+        this.liked = liked;
+        this.viewedList = currentUser.getViewed();
+        this.likedList = currentUser.getLiked();
+    }
+
+    /**
+     * Adds displayedUser to currentUser's viewed list.
+     * If liked is true, also adds displayedUser to currentUser's liked list.
+     * @param displayedUser is the user that is currently being displayed to currentUser.
+     * @param liked is True when currentUser 'likes' displayedUser, False otherwise.
+     */
+    public void addToList(User displayedUser, boolean liked) {
+        viewedList.add(displayedUser);
+        currentUser.setViewed(viewedList);
```

Comment:
I'm slightly confused as to what this line (line 35) does

Reply:
Line 35 is setting currentUser's viewed list to the currentUser's viewed list + the current displayedUser. I have both lines 34 and 35 because line 34 is just adding the current displayedUser to the viewedList (but doesn't update currentUser's viewed list), so that is what Line 35 is doing.